components/concurrency_manager/src/lock_table.rs

```diff
@@ -105,6 +105,15 @@ impl LockTable {
         None
     }
 
+    /// Iterates all handles and call a specified function on each of them.
+    pub fn for_each(&self, mut f: impl FnMut(Arc<KeyHandle>)) {
```

Comment:
Better to implement an Iterator for the lock table than add an ad hoc `for_each` method.